### ISSUE TITLE
.pt model and .onxx produce different results

### ISSUE THREAD

**Issue by jeff730** [sunsmarterjie/yolov12#152]
I have trained a custom model for barcode and QR code scanning. After several days of training and fine-tuning, the accuracy and logs look good.

When I use the .pt file, scanning works perfectly — both barcodes and QR codes are detected reliably.
However, after converting the model to ONNX (.onnx), the behavior changes:
Barcodes are still detected correctly.
QR codes are not detected at all.

I followed [this suggested solution](https://github.com/sunsmarterjie/yolov12/issues/90#issuecomment-2790453855)
but unfortunately it didn’t fix the issue for me. At this point, I’m stuck after investing a lot of time into training and fine-tuning.

Training Logs (last few epochs)
```
445,69408.3,0.17379,0.11333,0.79514,0.99616,1,0.995,0.94669,0.27302,0.13568,0.83703,4.96e-05,4.96e-05,4.96e-05
446,69903.4,0.17974,0.11613,0.80541,0.99606,1,0.995,0.94586,0.27317,0.13552,0.83819,4.3e-05,4.3e-05,4.3e-05
447,70384.2,0.17339,0.11126,0.80427,0.99604,1,0.995,0.95032,0.26605,0.13404,0.83501,3.64e-05,3.64e-05,3.64e-05
448,70875.4,0.17725,0.11309,0.80297,0.99601,1,0.995,0.94979,0.27131,0.1369,0.83644,2.98e-05,2.98e-05,2.98e-05
449,71376.8,0.16982,0.11079,0.80809,0.99605,1,0.995,0.94894,0.26827,0.1352,0.83493,2.32e-05,2.32e-05,2.32e-05
450,71875.1,0.17086,0.10893,0.79628,0.99602,1,0.995,0.95,0.26845,0.13686,0.83412,1.66e-05,1.66e-05,1.66e-05
```

Why does the .onnx model fail to detect QR codes while the .pt model works perfectly?
Is there something specific I need to check or modify during conversion?
Could this be related to export parameters, opset version, or post-processing differences between PyTorch and ONNX?

Any help or guidance would be greatly appreciated! 🙏
